src/units/types.cpp

```diff
@@ -887,8 +887,8 @@ void patch_movetype(
 	gui2::typed_formula<int> formula(formula_str);
 	wfl::map_formula_callable original;
 
-	boost::sregex_iterator m(formula_str.begin(), formula_str.end(), fai_identifier);
-	for(const boost::sregex_iterator::value_type& p : std::make_pair(m, boost::sregex_iterator())) {
+	std::sregex_iterator m(formula_str.begin(), formula_str.end(), fai_identifier);
+	for(const std::sregex_iterator::value_type& p : std::make_pair(m, std::sregex_iterator())) {
```

Comment:
I'm going to make a merge-conflict to your PR here, but the resolution is simply going to be removing this file from the PR.

Commit d922914a2042976cf17c9e5322b59d5c52b4eb95 took two bits of code, neither of which needed a regex, and combined them into a single block that did need a regex. Part of fixing #5308 will be reverting that.